
From: Paolo 'Blaisorblade' Giarrusso <blaisorblade_spam@yahoo.it>

This config option was lost during the creation of lib/Kconfig.debug, due to a
bad expressed dependency; I also moved the option back to its original place
for UML (it is near CONFIG_MCONSOLE since it depends on that).

Signed-off-by: Paolo 'Blaisorblade' Giarrusso <blaisorblade_spam@yahoo.it>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/arch/um/Kconfig   |   19 +++++++++++++++++++
 25-akpm/lib/Kconfig.debug |    1 -
 2 files changed, 19 insertions(+), 1 deletion(-)

diff -puN arch/um/Kconfig~uml-readd-config_magic_sysrq-for-uml arch/um/Kconfig
--- 25/arch/um/Kconfig~uml-readd-config_magic_sysrq-for-uml	Thu Jan 13 15:27:05 2005
+++ 25-akpm/arch/um/Kconfig	Thu Jan 13 15:27:05 2005
@@ -145,6 +145,25 @@ config MCONSOLE
 
         It is safe to say 'Y' here.
 
+config MAGIC_SYSRQ
+	bool "Magic SysRq key"
+	depends on MCONSOLE
+	---help---
+	If you say Y here, you will have some control over the system even
+	if the system crashes for example during kernel debugging (e.g., you
+	will be able to flush the buffer cache to disk, reboot the system
+	immediately or dump some status information). A key for each of the
+	possible requests is provided.
+
+	This is the feature normally accomplished by pressing a key
+	while holding SysRq (Alt+PrintScreen).
+
+	On UML, this is accomplished by sending a "sysrq" command with
+	mconsole, followed by the letter for the requested command.
+
+	The keys are documented in <file:Documentation/sysrq.txt>. Don't say Y
+	unless you really know what this hack does.
+
 config HOST_2G_2G
 	bool "2G/2G host address space split"
 	default n
diff -puN lib/Kconfig.debug~uml-readd-config_magic_sysrq-for-uml lib/Kconfig.debug
--- 25/lib/Kconfig.debug~uml-readd-config_magic_sysrq-for-uml	Thu Jan 13 15:27:05 2005
+++ 25-akpm/lib/Kconfig.debug	Thu Jan 13 15:27:05 2005
@@ -23,7 +23,6 @@ config MAGIC_SYSRQ
 config MAGIC_SYSRQ
 	bool "Magic SysRq key"
 	depends on DEBUG_KERNEL && (H8300 || M68KNOMMU || V850)
-	depends (USERMODE && MCONSOLE)
 	help
 	  Enables console device to interpret special characters as
 	  commands to dump state information.
_
